src/blocks/remote-data-no-result/edit.tsx

```diff
@@ -0,0 +1,48 @@
+/**
+ * WordPress dependencies
+ */
+import { useBlockProps, useInnerBlocksProps } from '@wordpress/block-editor';
+import { BlockEditProps, Template } from '@wordpress/blocks';
+
+import { useRemoteDataContext } from '@/blocks/remote-data-container/hooks/useRemoteDataContext';
+import { __ } from '@/utils/i18n';
+
+import './editor.scss';
+
+const NO_RESULTS_TEMPLATE: Template[] = [
+	[
+		'core/paragraph',
+		{
+			content: __( 'No results found.' ),
+		},
+	],
+];
+
+const NO_RESULTS_PLACEHOLDER_TEMPLATE: Template[] = [
+	[
+		'core/paragraph',
+		{
+			placeholder: __(
+				'This block only works when placed inside a remote data block when there are no results. This block will be ignored as currently configured.'
```

Comment:
Slight clarification since in the editor it doesn't matter if there are no results currently.

```suggestion
				'This block must be placed inside a remote data block. This block will be ignored as currently configured.'
```